src/pages/Account/ProfilView.jsx

```diff
@@ -74,24 +102,387 @@ export function ProfilView() {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: user.token
+      },
+      body: JSON.stringify(formData)
+    }).then((response) => {
+      if (response.ok) {
+        console.log('ok');
+        return true;
+      } else {
+        console.error('Error updating profile');
+        return false;
+      }
+    });
+
+  };
+
+  const openDeletePopup = () => {
+    setClickOnDelete(true);
+  }
+
+  const deleteProfile = () => {
+    handleDeleteUser().then((deleteIsValid) => {
+      if(!deleteIsValid) {
+        return;
+      }
+      
+      window.location.href = '/';
+      logout();
+    });
+  };
+
+  const handleDeleteUser = async () => {
+    return await fetch(`${apiURL}/users/${user.id}`, {
+      method: 'DELETE',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: user.token
+      }
+    }).then((response) => {
+      if (response.ok) {
+        return true;
+      } else {
+        console.error('Error deleting profile');
+        return false;
+      }
+    });
+  }
+
+  const cancelDeleteProfile = () => {
+    setClickOnDelete(false);
+  };
+
+  const changeFirstname = (firstname) => {
+    setFirstname(firstname);
+  };
+
+  const changeLastname = (firstname) => {
+    setLastname(firstname);
+  };
+
+  const changeEmail = (email) => {
+    setEmail(email);
+  };
+
+  const changePassword = (password) => {
+    setPassword(password);
+  };
+
+  const changeConfirmPassword = (confirmPassword) => {
+    setConfirmPassword(confirmPassword);
+  };
+
+  const totalAdventure = () => {
+    if(!userAdventureData) return 0;
+    return userAdventureData.length;
+  }
+
+  const adventurePercentageCompleted = () => {
+    //console.log(userAdventureData);
+    if(!userAdventureData) return 0;
+    let percentage = 0;
+    userAdventureData.map((adventure) => {
+      if(adventure.statut === 'termine') percentage++;
+    });
+
+    if(userAdventureData.length == 0) {
+      return 0;
+    }
+
+    return Math.floor((percentage/userAdventureData.length)*100);
+  };
+  
+  const booksFinished = () => {
+    if(!userAdventureData) return 0;
+    let finishedBooks = [];
+    userAdventureData.map((adventure) => {
+      if(adventure.statut === 'termine' && !finishedBooks.includes(adventure.id_livre)) finishedBooks.push(adventure.id_livre);
+    });
+    return finishedBooks.length;
+  }
+
+  const mostFavoriteBook = () => {
+    if(!userAdventureData || !books) return;
+    let mostFavoriteBooksId = [];
+    let mostFavoriteBooksTitle = [];
+    userAdventureData.map((adventure) => {
+      if(mostFavoriteBooksId[adventure.id_livre] === undefined) mostFavoriteBooksId[adventure.id_livre] = 1
+      else mostFavoriteBooksId[adventure.id_livre]++
+    });
+
+    const maxValue = Math.max(...Object.values(mostFavoriteBooksId));
+    const maxKeys = Object.keys(mostFavoriteBooksId).filter(key => mostFavoriteBooksId[key] === maxValue);
```

Comment:
Au lieu de faire ça tu devrait faire une liste d'objet et chaque objet tu as le nombre de book et l'id du book ça évite 5 for quoi ...